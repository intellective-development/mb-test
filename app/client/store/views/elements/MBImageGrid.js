// @flow

import * as React from 'react';
import cn from 'classnames';
import type { ImageType } from '../../business/content_module';
import { MBLayout } from '../elements';
import MBLink from './MBLink';
import MBHeader from './MBHeader';

export type ImageGridProps = {
  title: string,
  action_url: string,
  content: ImageType[],
}

type GridProps = {
  tiles: ImageType[]
}

type TileProps = {
  tile: ImageType
}

const ImageGrid = ({ content, title }: ImageGridProps) => {
  return (
    <MBLayout.StandardGrid cols={1} className="el-image-grid">
      <MBHeader title={title} />
      <Grid tiles={content} />
    </MBLayout.StandardGrid>
  );
};

const Grid = ({ tiles }: GridProps) => (
  <div className="el-image-grid__container">
    {tiles.map((tile) => (<Tile key={tile.internal_name} tile={tile} />))}
  </div>
);

const Tile = ({ tile }: TileProps) => (
  <MBLink.View
    key={tile.internal_name}
    className={cn('el-image-grid__tile')}
    href={tile.action_url}
    disabled={!tile.action_url}>
    <img
      alt={tile.internal_name}
      className="el-image-grid__image"
      src={tile.image_url} />
  </MBLink.View>
);

export default ImageGrid;
