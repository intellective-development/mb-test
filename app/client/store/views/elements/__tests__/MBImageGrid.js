
import * as React from 'react';
import MBImageGrid from '../MBImageGrid';

describe('MBImageGrid', () => {
  it('renders', () => {
    expect(shallow(<MBImageGrid title="foo" action_url="foo" content={[]} />)).toMatchSnapshot();
  });
});
