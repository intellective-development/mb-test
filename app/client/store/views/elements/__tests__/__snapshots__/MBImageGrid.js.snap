// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MBImageGrid renders 1`] = `
<StandardGrid
  className="el-image-grid"
  cols={1}
>
  <MBHeader
    title="foo"
  />
  <Grid
    tiles={Array []}
  />
</StandardGrid>
`;
